### ISSUE TITLE
figure out better vue router/link integration strategy

### ISSUE THREAD

**Issue by stasson** [stasson/vue-mdc-adapter#90]

**Comment by stasson** [comment]
- need @click dispatch for any component that may trigger routing
```html
<mdc-xxx  @click="$router.push('/route-xxx')" />
```
- need to add router link active and exact classes to active links (  drawer items, ... )

**Comment by griest024** [comment]
Is this just an issue with the demo or the project itself? Are you not happy with just using the `href`?

**Comment by stasson** [comment]
no the issue is the project itself: 

for clickable items ( buttons, drawer-items, links, etc), some people will want to use 
- an anchor with `href` 
- `@click=handler`
- router-link

we need to support all  3 ways. 

there is something interesting with router link : the [activeClass matching](
https://router.vuejs.org/en/api/options.html#linkexactactiveclass)

so for now I'm still trying the best way to support vue-router in a nice way: wrapper ? Mixin ? ...

**Comment by griest024** [comment]
How would you want these components to be used? Specify the link with a `link-to` prop and let the project decide the best way to implement it?

For example, if vue-router is not being used, `href` is really the only option. But since the vue-router docs specifically recommend `router-link` over an anchor, if vue-router is detected (`if (this.$router)`), we should implement the link with `router-link`.

In both cases we could add an event handler that emits a `link-clicked` event.

I think a mixin would definitely be the best option. If the user doesn't want to use any of our packaged solutions, they can always use a slot to fully customize the link.

**Comment by stasson** [comment]
@griest024 

So I played with router link today. at first I went for this : 
```html
<router-link><mdc-drawer-item /></router-link>
```
They say in the doc anchor children would get the right href applied:
- but it doesn't work with components v-nodes only with static elements 👎 
- anyway, not optimum to look for all children to find an anchor

So I decided to try the mixin way. I created a new [CustomLink](https://github.com/stasson/vue-mdc-adapter/blob/master/components/util/link.js) 
component which renders a router-link if there is a valid `to` property and if the vue-router plugin is registered. otherwise it falls back to anchor.

I ended up with this : 

### Item usage

- A simple link

```html
  <mdc-drawer-item href="#">Inbox</mdc-drawer-item>
```

- With vue-router

```html
  <mdc-drawer-item  to="/path">Inbox</mdc-drawer-item>
  <mdc-drawer-item  :to='folder' append >Inbox</mdc-drawer-item>
```

> Customize the active links style with vue-router active or exact-active classes:

```css
.mdc-drawer-item.router-link-exact-active {
  color: red;
}
```

- Trigger an event

```html
  <mdc-drawer-item event='my-event'>Inbox</mdc-drawer-item>
```

- Click handler

```html
  <mdc-drawer-item @click='handler'>Inbox</mdc-drawer-item>
```

Seems to work well. I guess we should apply same change for nav links components.
Wondering if we should do the same for buttons as well ...

**Comment by griest024** [comment]
@stasson for the item usage, wouldn't you rather have the user pass the link through a single prop as opposed to having to figure out which prop to use when?

**Comment by stasson** [comment]
closed by #199
